src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/UpdateSolicitorDetailsService.java

```diff
@@ -52,4 +68,51 @@ private Map<String, Object> convertOrganisationAddressToSolicitorAddress(Organis
             .postCode(organisationData.getContactInformation().get(0).getPostcode())
             .build(), Map.class);
     }
+
+    public Map<String, Object> updateSolicitorContactDetails(ChangedRepresentative addedSolicitor,
+                                                             Map<String, Object> caseData,
+                                                             boolean isConsented,
+                                                             boolean isApplicant) {
+        if (isApplicant) {
+            caseData.put(isConsented ? CONSENTED_SOLICITOR_NAME : CONTESTED_SOLICITOR_NAME,
+                addedSolicitor.getName());
+            caseData.put(isConsented ? SOLICITOR_EMAIL : CONTESTED_SOLICITOR_EMAIL, addedSolicitor.getEmail());
+            caseData.put(isConsented ? CONSENTED_SOLICITOR_FIRM : CONTESTED_SOLICITOR_FIRM,
+                addedSolicitor.getOrganisation().getOrganisationName());
+        } else {
+            updateRespSolFields(caseData, addedSolicitor);
+        }
+
+
+        return removeSolicitorFields(caseData, isConsented, isApplicant);
```

Comment:
this is a code smell. you should have this method call after updateSolicitor is called not a chain method call inside of it.